### PR TITLE
Refactor out shape validation to be more generically useful

### PR DESCRIPTION
**Pull request by mdales**
I needed to have the check_and_format_shape method be accessible from other code, so I moved it to `data/gedi_download_pipeline.py` and removed other code as discussed. Over time I'll add more there  I suspect.

I also added some tests, as I needed to remove side-effects from the function and keep them in the original file, and I wanted some confidence I hadn't broken too much.

### CODE REVIEW

File: src/spark/gedi_download_pipeline.py [hunk 1]
```diff
@@ -348,7 +298,23 @@ def exec_spark(
...
+        try:
+            shp = check_and_format_shape(shp)
+        except DetailError:
+            input(
+                (
+                    "The NASA API can only accept up to 5000 vertices in a single shape,\n"
+                    "but the shape you supplied has {} vertices.\n"
+                    "If you would like to automatically simplify this shape to its\n"
+                    "bounding box, press ENTER, otherwise Ctrl-C to quit."
+                ).format(n_coords)
```
Comment (ameliaholcomb) [inline src/spark/gedi_download_pipeline.py]: I think this variable is no longer in scope?

File: tests/data/test_inputs.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+def test_too_many_points_simplify():
+	points = []
+	for i in range(5001):
+		points.append((0.1, 0.0001 * i))
+	polygon = Polygon(points)
+	geometry = gpd.GeoSeries(polygon)
+
+	checked = check_and_format_shape(geometry, simplify=True)
+	assert checked is not None
```
Comment (ameliaholcomb) [inline tests/data/test_inputs.py]: Maybe better to do:
```python
assert checked.contains(geometry)
```

File: src/spark/gedi_download_pipeline.py (hunk 1 above)
Comment (mdales) [reply src/spark/gedi_download_pipeline.py]: Good catch! Made it a parameter on the exception so we can still display list.

File: tests/data/test_inputs.py [hunk 3]
```diff
@@ -65,4 +65,4 @@ def test_too_many_points_simplify():
 	geometry = gpd.GeoSeries(polygon)
 
 	checked = check_and_format_shape(geometry, simplify=True)
-	assert checked is not None
+	checked.contains(geometry)
```
Comment (ameliaholcomb) [inline tests/data/test_inputs.py]: assert True?

Comment (ameliaholcomb) [reply tests/data/test_inputs.py]: I think it's like 
```python
assert checked.contains(geometry).all()
```

Comment (mdales) [reply tests/data/test_inputs.py]: doh, that's what I get for rushing. I also had to make a more meaningful geometry, as the check you suggested didn't work initially as I just made a straight line :)

**Review by ameliaholcomb — APPROVED** [review]
LGTM, thank you!

**Pull request merged**